.github/workflows/ci-pr-validation.yaml

```diff
@@ -219,13 +219,50 @@ jobs:
       - name: Build packages
         run: pkg/${{matrix.pkg.type}}/docker-build-${{matrix.pkg.type}}-${{matrix.cpu.platform}}.sh build:latest
 
+  cpp-build-macos:
+    timeout-minutes: 120
+    name: Build CPP Client on macOS
+    runs-on: macos-12
+    needs: unit-tests
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+
+      - name: Install dependencies
+        run: brew install openssl protobuf boost boost-python3 googletest zstd snappy
+
+      - name: Configure (default)
+        shell: bash
+        run: |
+          cmake \
+              -B ./build-macos \
+              -DBUILD_TESTS=OFF \
+              -S .
+
+      - name: Compile
+        shell: bash
+        run: |
+          cmake --build ./build-macos --parallel --config Release
+
+      - name: Configure (dynamic library only)
+        shell: bash
+        run: |
+          cmake \
+              -B ./build-macos-dynamic-library \
+              -DBUILD_TESTS=OFF \
+              -DBUILD_STATIC_LIB=OFF \
+              -S .
+
+      - name: Compile
+        shell: bash
+        run: |
+          cmake --build ./build-macos-dynamic-library --parallel --config Release
```

Comment:
I forgot why did I add a test for `-DBUILD_STATIC_LIB=OFF` option for Windows build. But I think there is no need to test this option now.

Reply:
Fixed